src/Sulu/Bundle/WebsiteBundle/Entity/AnalyticsRepository.php

```diff
@@ -19,22 +19,25 @@
 class AnalyticsRepository extends EntityRepository
 {
     /**
-     * Returns list of analytics.
+     * Returns list of analytics filterd by webspace key and environment.
      *
-     * @param $webspaceKey
+     * @param string $webspaceKey
+     * @param string $environment
      *
      * @return Analytics[]
      */
-    public function findByWebspaceKey($webspaceKey)
+    public function findByWebspaceKeyAndEnvironment($webspaceKey, $environment)
```

Comment:
undocumented BC Break. Add a line to the `UPGRADE.md` or avoid the bcbreak:

To avoid the bc break we could keep the old name and make the $environment optional 

```php
public function findByWebspaceKey($webspaceKey, $environment = null)
          // ...

          if ($environment) {
              $queryBuilder->andWhere('a.allDomains = TRUE OR domains.environment = :environment');
              $queryBuilder->setParameter('environment', $environment);
          } else {
              @trigger_error('Calling findByWebspace without environment is deprecated and will maybe return a false result');
          }

          // ...
}
```

/cc what you think @danrot

Reply:
Why not using setter injection to set the environment? As a developer using this class I ideally do not have to worry about the environment.

Reply:
@danrot sounds like a good idea:

@trickreich you can call the setter in the service definition:

```
        <service id="sulu_website.analytics.repository"
                 class="Sulu\Bundle\WebsiteBundle\Entity\AnalyticsRepository">
            <factory service="doctrine.orm.entity_manager" method="getRepository"/>

            <argument type="string">SuluWebsiteBundle:Analytics</argument>
            
            <call method="setEnvironment">
                <argument>%kernel.environment%</argument>
            </call>
        </service>
```